bin/orcabus.ts

```diff
@@ -27,12 +27,13 @@ const props: cdk.StackProps = {
 const config = getEnvironmentConfig('beta');
 if (!config) throw new Error('No Config');
 
-new OrcaBusStatefulStack(app, 'OrcaBusStatefulStack', {
+const statefulStack = new OrcaBusStatefulStack(app, 'OrcaBusStatefulStack', {
   ...config.stackProps.orcaBusStatefulConfig,
   ...props,
 });
 
 new OrcaBusStatelessStack(app, 'OrcaBusStatelessStack', {
+  statefulStack,
```

Comment:
The stateless stack should not depend on the stateful stack. 
Ideally there are only static parameters passed between those stacks to maintain independence. 

If I see this right, the only stateful resource that needs to be passed to the stateless stack is the SQS Queue, which should be possibly via it's ARN, no?

Reply:
Yes, could de-couple this more. An ARN should be enough to reconstruct the `IQueue` construct using [`fromQueueArn`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_sqs.Queue.html#static-fromwbrqueuewbrarnscope-id-queuearn).

Although isn't this still the stateless stack depending on the stateful stack, just more indirected? Is the advantage that it isn't the whole stateful stack, just the required resource?

Reply:
It's less a full stack vs a single resource thing, it's more that there is now a dependency between those stacks that potentially can cause issues down the road.
(also see: https://www.endoflineblog.com/cdk-tips-03-how-to-unblock-cross-stack-references)

Also, with this setup you (in this case unnecessarily) force the stateless stack to depend on another CDK stack/construct. What if I want to create the SQS via Terraform? 

Ideally, the stateless stack should just take a list of (string) parameters and be agnostic to how those stateful resources are created/managed.

Reply:
Yes, that's a good point. So in this case, the string parameter would be an ARN for the SQS queue, right?